frontend/src/views/c-summary.vue

```diff
@@ -127,38 +127,50 @@
   )
 </template>
 
-<script>
+<script lang='ts'>
 import { mapState } from 'vuex';
+import { PropType, defineComponent } from 'vue';
 
 import cSummaryCharts from '../components/c-summary-charts.vue';
 import getNonRepeatingColor from '../utils/ramp-colour-generator';
 import sortFiltered from '../utils/repo-sorter';
+import {
+  Commit,
+  DailyCommit,
+  Repo,
+  User,
+  isCommit,
+  CommitResult,
+} from '../types/types';
+import { ErrorMessage } from '../types/zod/summary-type';
+import { AuthorFileTypeContributions, FileTypeAndContribution } from '../types/zod/commits-type';
+import { ZoomInfo } from '../types/vuex.d';
 
 const getFontColor = window.getFontColor;
 const dateFormatRegex = /([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))$/;
 
-export default {
+export default defineComponent({
   name: 'c-summary',
   components: {
     cSummaryCharts,
   },
   props: {
     repos: {
-      type: Array,
+      type: Array as PropType<Repo[]>,
       required: true,
     },
     errorMessages: {
-      type: Object,
+      type: Object as PropType<ErrorMessage>,
       default() {
         return {};
       },
     },
   },
   data() {
     return {
-      checkedFileTypes: [],
-      fileTypes: [],
-      filtered: [],
+      checkedFileTypes: [] as string[],
+      fileTypes: [] as string[],
+      filtered: [] as User[][],
       filterSearch: '',
       filterGroupSelection: 'groupByRepos',
```

Comment:
Do you think it would be useful to add `enum`s for typing the different selection choices we have in the summary-picker dropdowns?

Reply:
Great point, I added in the enums in the latest commit(s)!